internal/encryption/registry.go

```diff
@@ -0,0 +1,30 @@
+package encryption
+
+// Registry is a holder of KeyProvider and Method implementations. Key providers and methods can register
+// themselves with this registry.
+type Registry struct {
+	KeyProviders map[string]KeyProviderSource
+	Methods      map[string]MethodSource
+}
+
+func NewRegistry() Registry {
+	return Registry{
+		KeyProviders: map[string]KeyProviderSource{
+			"passphrase": func() KeyProvider { return &PassphraseKeyProvider{} },
+		},
+		Methods: map[string]MethodSource{
+			"aes_cipher": func() Method { return &AESCipherMethod{} },
+		},
+	}
+}
+
+type KeyProvider interface {
+	KeyData() ([]byte, error)
```

Comment:
How will a key provider place information in the "encryption" JSON key, such as a nonce used to derive the key from its input?

Compare the KeyProvider interface in https://github.com/opentofu/opentofu/pull/1132/files#diff-e7749ff1b789f4930684fd4132a65fe3227b0a2dcbb23fe80794ddd5066cd88b this is why the parameters include an EncryptionInfo block, which later gets encoded into the "encryption" JSON substructure.